resolver/sudn_resolver.go

```diff
@@ -0,0 +1,141 @@
+package resolver
+
+import (
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/0xERR0R/blocky/model"
+	"github.com/miekg/dns"
+)
+
+const (
+	sudnTest      = "test."
+	sudnInvalid   = "invalid."
+	sudnLocalhost = "localhost."
+)
+
+func sudnArpaSlice() []string {
+	return []string{
+		"10.in-addr.arpa.",
+		"21.172.in-addr.arpa.",
+		"26.172.in-addr.arpa.",
+		"16.172.in-addr.arpa.",
+		"22.172.in-addr.arpa.",
+		"27.172.in-addr.arpa.",
+		"17.172.in-addr.arpa.",
+		"30.172.in-addr.arpa.",
+		"28.172.in-addr.arpa.",
+		"18.172.in-addr.arpa.",
+		"23.172.in-addr.arpa.",
+		"29.172.in-addr.arpa.",
+		"19.172.in-addr.arpa.",
+		"24.172.in-addr.arpa.",
+		"31.172.in-addr.arpa.",
+		"20.172.in-addr.arpa.",
+		"25.172.in-addr.arpa.",
+		"168.192.in-addr.arpa.",
+	}
+}
+
+type SudnResolver struct {
+	NextResolver
+}
+
+func NewSudnResolver() ChainedResolver {
+	return &SudnResolver{}
+}
+
+func (r *SudnResolver) Resolve(request *model.Request) (*model.Response, error) {
+	if r.isSpecial(request, sudnArpaSlice()...) ||
+		r.isSpecial(request, sudnInvalid) ||
+		r.isSpecial(request, sudnTest) {
+		return r.negativeResponse()
+	} else if r.isSpecial(request, sudnLocalhost) {
+		qtype := request.Req.Question[0].Qtype
+		fmt.Println("QType:", qtype)
+		switch qtype {
+		case dns.TypeA:
+			return r.loopbackResponseA()
+		case dns.TypeAAAA:
+			return r.loopbackResponseAAAA()
+		default:
+			return r.negativeResponse()
+		}
+	}
+
+	return r.next.Resolve(request)
+}
+
+func (r *SudnResolver) Configuration() []string {
+	return []string{"Special-Use Domain Names (RFC 6761)"}
```

Comment:
This could probably be covered in a test, as for example here: https://github.com/0xERR0R/blocky/blob/48dbc5bc13d1d72f1a42df85c058c0559d1f5ba5/resolver/custom_dns_resolver_test.go#L195

Reply:
Yea it would actually be a pretty silly test as it is a fixed string and the methode is only included to qualify as ChainedResolver.
I will add it anyway 😅